sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -717,7 +717,8 @@ object Limit {
   }
 }
 
-case class GlobalLimit(limitExpr: Expression, child: LogicalPlan) extends UnaryNode {
+abstract class LimitNode extends UnaryNode {
+  def limitExpr: Expression
```

Comment:
do we assume `limitExpr` is foldable? but seems there is no type checking logic for it.